event/event.go

```diff
@@ -42,9 +42,27 @@ type (
 	// by using [slog.FromCtx].
 	Handler[T any] func(context.Context, T) error
 
+	// HandlerWithMetadata is responsible for handling events from a [Subscription] with the associated event metadata.
+	// The context passed to the handler will have all common metadata also passed to calls to [Handler] and extra metadata
+	// that is more specific as defined on [Metadata].
+	HandlerWithMetadata[T any] func(context.Context, T, Metadata) error
+
 	// Message represents a raw message received on a subscription.
 	Message struct {
-		body []byte
+		Body     []byte
```

Comment:
Since slices are a nightmare of mutability I think having a method added too much complexity without bringing too much benefits =/ (Im excited about avoiding mutability...but in Go it is not very easy to do that...)